pyttb/gcp/fg.py

```diff
@@ -74,6 +74,9 @@ def evaluate(
             "Either a function handle, or a gradient handle must be provided."
         )
 
+    if isinstance(data, ttb.sptensor):
+        data = data.double()
```

Comment:
This should be `data = data.full()` to keep things consistent/related to my comment below. With that change you shouldn't need all the numpy operation changes